src/scripts.js

```diff
@@ -80,23 +98,60 @@ function assignCurrentCustomer(customerDataset, userID) {
          currentCustomer = new Customer(customer)
         }
     })
-    domUpdates.toggleHidden(loginWrapper, customerDetailsWrapper)
+    domUpdates.toggleHidden(loginWrapper)
+    domUpdates.toggleHidden(customerDetailsWrapper)
+    domUpdates.toggleHidden(bookButton)
     domUpdates.welcomeCustomer(currentCustomer.name)
     domUpdates.displayCustomerData(ledger, hotel, currentCustomer)
 }
 
-function fillHotel(roomsDataset) {
-    roomsDataset.forEach(room => hotel.push(room))
+function displayPastTrips() {
+    currentCustomer.sortBookings()
+    domUpdates.renderTrips(roomDisplayWrapper, currentCustomer.returnPastTrips(today))
 }
 
-function fillLedger(bookingsDataset) {
-    bookingsDataset.forEach(booking => ledger.push(booking))
+function displayFutureTrips() {
+    currentCustomer.sortBookings()
+    domUpdates.renderTrips(roomDisplayWrapper, currentCustomer.returnFutureTrips(today))
 }
 
-function displayPastTrips() {
-    domUpdates.renderPastTrips(roomDisplayWrapper, currentCustomer.bookings)
+function displayDatePicker() {
+    domUpdates.toggleHidden(customerDetailsWrapper)
+    domUpdates.toggleHidden(calendarWrapper)
+    calendar.value = today
+}
+
+function findRoomsByType(potentialRooms, type) {
+    let matchedRooms = potentialRooms.filter(room => room.roomType === type)
+
+    if(matchedRooms.length) {
+        return matchedRooms
+    }else {
+        return false
+    }
 }
 
+function findAvailableRooms() {
+    let desiredDate = calendar.value
+    let scrubbedDate = desiredDate.split('-').join('/')
+
+        let unBooked = ledger.filter(booking => {
+            return booking.date === scrubbedDate
+        }).map(booking => booking.roomNumber)
+
+    let availableRooms = hotel.filter(room => !unBooked.includes(room.number))
+    console.log(availableRooms)
 
```

Comment:
You can make your filter selection mandatory by adding an if statement in this function, if the value of typeFilter is 'empty', you want to display an error message without submitting. Alternately, you could make an 'all rooms' option and show them all by default when the form is submitted. Might be good for user experience!